src/rpc_client/mod.rs

```diff
@@ -337,46 +343,43 @@ impl EthRpcClient {
             _ => 24 * 1024, // Default for unknown networks
         };
 
-        let results: MultiCallResults<Block> = self
-            .parallel_call(
-                "eth_getBlockByNumber",
-                GetBlockByNumberParams {
-                    block,
-                    include_full_transactions: false,
-                },
-                self.response_size_estimate(expected_block_size + HEADER_SIZE_LIMIT),
-            )
-            .await;
-        results.reduce_with_equality()
+        self.parallel_call(
+            "eth_getBlockByNumber",
+            GetBlockByNumberParams {
+                block,
+                include_full_transactions: false,
+            },
+            self.response_size_estimate(expected_block_size + HEADER_SIZE_LIMIT),
+        )
+        .await
+        .reduce(self.consensus_strategy())
     }
 
     pub async fn eth_get_transaction_receipt(
         &self,
         tx_hash: Hash,
     ) -> Result<Option<TransactionReceipt>, MultiCallError<Option<TransactionReceipt>>> {
-        let results: MultiCallResults<Option<TransactionReceipt>> = self
-            .parallel_call(
-                "eth_getTransactionReceipt",
-                vec![tx_hash],
-                self.response_size_estimate(700 + HEADER_SIZE_LIMIT),
-            )
-            .await;
-        results.reduce_with_equality()
+        self.parallel_call(
+            "eth_getTransactionReceipt",
+            vec![tx_hash],
+            self.response_size_estimate(700 + HEADER_SIZE_LIMIT),
+        )
+        .await
+        .reduce(self.consensus_strategy())
     }
 
     pub async fn eth_fee_history(
         &self,
         params: FeeHistoryParams,
     ) -> Result<FeeHistory, MultiCallError<FeeHistory>> {
         // A typical response is slightly above 300 bytes.
-        let results: MultiCallResults<FeeHistory> = self
-            .parallel_call(
-                "eth_feeHistory",
-                params,
-                self.response_size_estimate(512 + HEADER_SIZE_LIMIT),
-            )
-            .await;
-        results.reduce_with_strict_majority_by_key(|fee_history| fee_history.oldest_block)
```

Comment:
According to what we discussed on Slack, it doesn't make sense for `eth_fee_history` to use a separate dedicated strategy.